app/components/selection/handle.element.js

```diff
@@ -0,0 +1,184 @@
+					const newHeight = initialHeight - diffY
+					const newTranslate = initialTransform.translate(0, diffY).transformPoint()
+
+					requestAnimationFrame(() => {
+						sourceEl.style.height = `${newHeight}px`
+						sourceEl.style.transform = `translate(${newTranslate.x}px, ${newTranslate.y}px)`
+					})
+					break
+				}
+				case 'top-end': {
+					const newWidth = initialWidth + diffX
+					const newHeight = initialHeight - diffY
+					const newTranslate = initialTransform.translate(0, diffY).transformPoint()
+
+					requestAnimationFrame(() => {
+						sourceEl.style.width = `${newWidth}px`
+						sourceEl.style.height = `${newHeight}px`
+						sourceEl.style.transform = `translate(${newTranslate.x}px, ${newTranslate.y}px)`
+					})
+					break
+				}
+				case 'middle-start': {
+					const newWidth = initialWidth - diffX
+					const newTranslate = initialTransform.translate(diffX).transformPoint()
+
+					requestAnimationFrame(() => {
+						sourceEl.style.width = `${newWidth}px`
+						sourceEl.style.transform = `translate(${newTranslate.x}px, ${newTranslate.y}px)`
+					})
+					break
+				}
+				case 'middle-end': {
+					const newWidth = initialWidth + diffX
+
+					requestAnimationFrame(() => {
+						sourceEl.style.width = `${newWidth}px`
+					})
+					break
+				}
+				case 'bottom-start': {
+					const newWidth = initialWidth - diffX
+					const newHeight = initialHeight + diffY
+					const newTranslate = initialTransform.translate(diffX, 0).transformPoint()
+
+					requestAnimationFrame(() => {
+						sourceEl.style.width = `${newWidth}px`
+						sourceEl.style.height = `${newHeight}px`
+						sourceEl.style.transform = `translate(${newTranslate.x}px, ${newTranslate.y}px)`
+					})
+					break
+				}
+				case 'bottom-center': {
+					const newHeight = initialHeight + diffY
+
+					requestAnimationFrame(() => {
+						sourceEl.style.height = `${newHeight}px`
+					})
+					break
+				}
+				case 'bottom-end': {
+					const newWidth = initialWidth + diffX
+					const newHeight = initialHeight + diffY
+
+					requestAnimationFrame(() => {
+						sourceEl.style.width = `${newWidth}px`
+						sourceEl.style.height = `${newHeight}px`
+					})
+					break
+				}
+			}
+		}
+
+		document.addEventListener('pointerup', on_element_resize_end, { once: true })
+		document.addEventListener('mouseleave', on_element_resize_end, { once: true })
+
+		function on_element_resize_end() {
+			document.removeEventListener('pointermove', on_element_resize_move)
+			document.body.style.cursor = originalDocumentCursor
+			document.body.style.userSelect = originalDocumentUserSelect
+			sourceEl.style.transition = originalElTransition
+		}
+	}
+
+	disconnectedCallback() {
+		this.button.removeEventListener('pointerdown', this.on_element_resize_start.bind(this))
+	}
+
+	render() {
+		return `
+			<button type="button" aria-label="Resize"></button>
+		`
+	}
+}
+
+customElements.define('visbug-handle', Handle)
+
+// utility functions
```

Comment:
utility worked out great! might be worth putting [into here](https://github.com/GoogleChromeLabs/ProjectVisBug/blob/main/app/utilities/numbers.js) for any future use?

Reply:
Done! https://github.com/GoogleChromeLabs/ProjectVisBug/pull/569/commits/ec8d08ad7c754743acdc3e834e6d21cbba8437d7